### PR TITLE
[Editor] Add gutter hover effect

### PR DESCRIPTION
**Pull request by Sekhmet**
Fixes Issue: #5603 

### Summary of Changes

* Add hover effect to gutter line numbers.

### Test Plan

- [x] Open JS file.
- [x] Hover on the gutter line numbers.
- [x] Line numbers should be highlighted.
- [x] Empty lines shouldn't be highlighted.
- [x] Existing breakpoints shouldn't be highlighted.
- [x] Highlight should disappear after leaving gutter area.

### Screenshots/Videos

Before:
gutter_before

After:
gutter_after

### CODE REVIEW

**Comment by jasonLaster** [comment]
Nice. Does this work when a file has 100+ or 1000+ lines?

**Comment by Sekhmet** [comment]
I couldn't reproduce this problem (both on Firefox and Chrome on Linux). Is there anything specific that might cause this?
Maybe there is something that changes line number color, if it's the case does that patch solve the problem?
```diff
diff --git a/src/components/Editor/Editor.css b/src/components/Editor/Editor.css
index 12f2ed8..68f924f 100644
--- a/src/components/Editor/Editor.css
+++ b/src/components/Editor/Editor.css
@@ -65,7 +65,7 @@ html[dir="rtl"] .editor-mount {
 :not(.empty-line):not(.new-breakpoint)
   > .CodeMirror-gutter-wrapper:hover
   > .CodeMirror-linenumber {
-  color: white;
+  color: white !important;
   left: auto !important;
   right: 4px;
 }
```


What should transition apply to? Background and text color?

**Comment by bomsy** [comment]
Random Q! should the arrow head pop of the edge of the gutter, and maybe the bottom of the triangle should line up with the edge of the gutter.

**Comment by Sekhmet** [comment]
That's something that Jason pointed out as well. Right now hover mark is styled the same way as regular breakpoint. I can update both of them in this PR. It should look better that way.

**Comment by jasonLaster** [comment]
@Sekhmet can you try styling it so on hover we get the same soft grey hover (w/o a bp) as we get when hovering a button? Curious how that would look

**Comment by jasonLaster** [comment]
I started a transparent hack here 
https://gist.github.com/56463fa5ae15cf72a4ca0b281a40ed55

i couldnt figure out how to show the line number...

**Comment by Sekhmet** [comment]
Grey background:
gutter_grey

Grey background withour icon:
gutter_grey_without_arrow

**Comment by Sekhmet** [comment]
Regular version with opacity removed and icon aligned to the edge of gutter:
gutter_normal_aligned

**Comment by darkwing** [comment]
@jasonLaster If you like grey (I think I like that more than blue), this var we're already using would work:  

```css
.editor.new-breakpoint svg {
  fill: var(--theme-highlight-gray)
}

```

hover

**Comment by jasonLaster** [comment]
yeah, i think the grey might be nice

**Comment by jasonLaster** [comment]
This is looking really good. 

- lets get circle green
- perhaps lets darken the highlighted line number to give some pop

File: src/components/Editor/EmptyLines.css [hunk 1]
```diff
@@ -1,7 +1,7 @@
 /* This Source Code Form is subject to the terms of the Mozilla Public
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at <http://mozilla.org/MPL/2.0/>. */
-
+/*
```
Comment (darkwing) [inline src/components/Editor/EmptyLines.css]: I believe we still want this, right?

File: src/components/Editor/Editor.css [hunk 2]
```diff
@@ -51,6 +51,23 @@ html[dir="rtl"] .editor-mount {
   direction: ltr;
 }
 
+:not(.empty-line):not(.new-breakpoint) > .CodeMirror-gutter-wrapper:hover {
+  width: 60px !important;
```
Comment (darkwing) [inline src/components/Editor/Editor.css]: I'm slightly concerned with all of the `!important` declarations -- can we adjust our selector specificity so we don't need them?

File: src/components/Editor/EmptyLines.css (hunk 1 above)
Comment (jasonLaster) [reply src/components/Editor/EmptyLines.css]: oh... lets rm this file

File: src/components/Editor/Editor.css (hunk 2 above)
Comment (jasonLaster) [reply src/components/Editor/Editor.css]: this might be hard to fix.. but yes, we should try

File: src/components/Editor/EmptyLines.css (hunk 1 above)
Comment (darkwing) [reply src/components/Editor/EmptyLines.css]: Sorry!  I wasn't aware -- remove away! :)

File: src/components/Editor/Editor.css (hunk 2 above)
Comment (jasonLaster) [reply src/components/Editor/Editor.css]: i see. makes sense

File: src/components/Editor/Editor.css [hunk 3]
```diff
@@ -51,6 +51,22 @@ html[dir="rtl"] .editor-mount {
...
+  background-color: var(--theme-highlight-gray) !important;
+  mask: url(/images/breakpoint.svg) no-repeat;
+  mask-size: 100%;
+  mask-position: 0 1px;
+}
+
+:not(.empty-line):not(.new-breakpoint) > .CodeMirror-gutter-wrapper:hover > .CodeMirror-linenumber {
+  left: auto !important;
+  right: 6px;
+  color: var(--theme-body-color);
```
Comment (Sekhmet) [inline src/components/Editor/Editor.css]: This looks good on light theme, but on dark theme it's hard to read. Is there a color that might work well for this?

**Comment by jasonLaster** [comment]
hmm, we still want to prevent highlighting empty lines

File: src/components/Editor/Editor.css (hunk 3 above)
Comment (darkwing) [reply src/components/Editor/Editor.css]: I love the detail in testing the dark theme too! Great work!  👍 

How about `--theme-body-color-alt` for the dark theme?  Seem reasonable to me :)

**Comment by violasong** [comment]
It's helpful to see all the variations of this - thanks all! One idea - Could we try using the row-hover background colors from the inspector light and dark themes? (It's a very subtle blue - maybe too subtle.) White-on-gray could be good for the disabled breakpoints.

**Comment by jasonLaster** [comment]
@violasong i went with a solid grey as it's the least intrusive now:

- row-hover had opacity which meant it'd clip
- disable bps is light blue and changing it seems like something we can do once this lands and we know we like it...

**Comment by violasong** [comment]
Ah yes, that makes sense to keep disable bps the existing light blue for now - sounds good!

**Pull request merged**